### PR TITLE
Scan formatter backticks patch

### PR DESCRIPTION
**Pull request by wag-miles**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description

Fixed the 'formatter' argument for scan using backticks to quote the value. Couldn't run scan with a custom formatter.

### CODE REVIEW

**Comment by wag-miles** [comment]
Signed

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Can you remove the changes to ` fastlane/spec/fixtures/xcodeproj/automatic_code_signing.xcodeproj/project.pbxproj`?

This change is a side effect of running `rspec` but a fix for this was just merged in so this won't be an issue in the future 😊

**Comment by wag-miles** [comment]
Done

File: scan/lib/scan/test_command_generator.rb [hunk 1]
```diff
@@ -85,9 +85,9 @@ def pipe
 
       formatter = []
       if Scan.config[:formatter]
-        formatter << "-f `#{Scan.config[:formatter]}`"
+        formatter << "-f '#{Scan.config[:formatter]}'"
       elsif FastlaneCore::Env.truthy?("TRAVIS")
-        formatter << "-f `xcpretty-travis-formatter`"
+        formatter << "-f 'xcpretty-travis-formatter'"
```
Comment (joshdholtz) [inline scan/lib/scan/test_command_generator.rb]: The backticks are actually needed if you are loading a custom formatter as a gem (which is what this is doing so this needs to stay as backticks)

File: scan/lib/scan/test_command_generator.rb [hunk 2]
```diff
@@ -85,9 +85,9 @@ def pipe
 
       formatter = []
       if Scan.config[:formatter]
-        formatter << "-f `#{Scan.config[:formatter]}`"
+        formatter << "-f '#{Scan.config[:formatter]}'"
```
Comment (joshdholtz) [inline scan/lib/scan/test_command_generator.rb]: Single quotes works if `:formatter` is a file path but this also needs to be backticks if `:formatter` is a gem 

I _think_ what we need to do is only use single quotes **if** `:formatter` end in `.rb` and if the file exists 🤔

**Review by joshdholtz — CHANGES_REQUESTED** [review]

**Comment by wag-miles** [comment]
I applied the latest suggestions and created an additional test case

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Can you run rubocop on this so it passes the linter? 😊

**Comment by wag-miles** [comment]
Done. Aggressive formatter on the loose

**Review by joshdholtz — APPROVED** [review]
Looks good! Thanks for doing the changes 😊

**Pull request merged**